app/views/ShareMessageView/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useLayoutEffect, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+import { StackNavigationOptions } from '@react-navigation/stack';
+import { RouteProp, StackActions, useNavigation, useRoute } from '@react-navigation/native';
+
+import { getPermalinkMessage } from '../../lib/methods';
+import { TGetCustomEmoji } from '../../definitions';
+import KeyboardView from '../../containers/KeyboardView';
+import scrollPersistTaps from '../../lib/methods/helpers/scrollPersistTaps';
+import I18n from '../../i18n';
+import * as HeaderButton from '../../containers/HeaderButton';
+import StatusBar from '../../containers/StatusBar';
+import { useTheme } from '../../theme';
+import { getUserSelector } from '../../selectors/login';
+import SafeAreaView from '../../containers/SafeAreaView';
+import { events, logEvent } from '../../lib/methods/helpers/log';
+import styles from './styles';
+import SelectPersonOrChannel from './SelectPersonOrChannel';
+import { useAppSelector } from '../../lib/hooks';
+import Message from '../../containers/message';
+import { NewMessageStackParamList } from '../../stacks/types';
+import { postMessage } from '../../lib/services/restApi';
+
+const ShareMessageView = () => {
+	const [roomsId, setRoomsId] = useState<string[]>([]);
+	const [sending, setSending] = useState(false);
+	const navigation = useNavigation();
+	const { theme, colors } = useTheme();
+
+	const {
+		params: { message }
+	} = useRoute<RouteProp<NewMessageStackParamList, 'ShareMessageView'>>();
+
+	const { blockUnauthenticatedAccess, server, serverVersion, user, baseUrl, Message_TimeFormat, customEmojis } = useAppSelector(
+		state => ({
+			user: getUserSelector(state),
+			server: state.server.server,
+			blockUnauthenticatedAccess: !!state.settings.Accounts_AvatarBlockUnauthenticatedAccess ?? true,
+			serverVersion: state.server.version as string,
+			baseUrl: state.server.server,
+			Message_TimeFormat: state.settings.Message_TimeFormat as string,
+			customEmojis: state.customEmojis
+		})
+	);
+
+	useLayoutEffect(() => {
+		const isSendButtonEnabled = roomsId.length > 0 && !sending;
+		navigation.setOptions({
+			title: I18n.t('Share_message'),
+			headerRight: () => (
+				<HeaderButton.Container>
+					<HeaderButton.Item
+						title={I18n.t('Send')}
+						color={isSendButtonEnabled ? colors.actionTintColor : colors.headerTintColor}
+						disabled={!isSendButtonEnabled}
+						onPress={handlePostMessage}
+						testID='share-message-view-send'
+					/>
+				</HeaderButton.Container>
+			),
+			headerLeft: () => <HeaderButton.CloseModal navigation={navigation} />
+		} as StackNavigationOptions);
+	}, [roomsId.length, navigation, sending]);
+
+	const handlePostMessage = async () => {
+		setSending(true);
+		const permalink = await getPermalinkMessage(message);
+		const msg = `[ ](${permalink})\n`;
+		await Promise.all(roomsId.map(roomId => postMessage(roomId, msg)));
```

Comment:
This comment gives me the impression this loop isn't really needed, but I see the frontend is doing the same thing 🙈 https://github.com/RocketChat/Rocket.Chat/blob/67a299b3546d702620c17b72cb81e31981b96de5/apps/meteor/app/api/server/v1/chat.ts#L206